apps/yggdrasil/lib/yggdrasil/user.ex

```diff
@@ -10,6 +14,8 @@ defmodule Yggdrasil.User do
     field :password, :string, virtual: true # not part of table
     field :password_confirmation, :string, virtual: true # not part of table
 
+    has_many :user_roles, Yggdrasil.UserRole
```

Comment:
In ecto 2.0 this can be roles, for now though it probably should be UserRoles otherwise the use is awkward I think if it was called roles then for example iterating through it would look like this

```elixir
Enum.map user.roles, fn r ->
  # it's called role, but it's not a role, I have to dot into the actual role so it's role.role?
  r.role
end
```

In ecto 2.0, we can eliminate the join table and do this

```elixir
many_to_many :roles, Yggdrasil.Role, join_through: "users_roles"

# then this is more natural
Enum.map user.roles, fn r ->
  r.resource # etc...
end
```